### PR TITLE
tablet_allocator: Change decomission failure message 

### LINKED ISSUE #20417: Unclear error message in case a node decommission is not allowed with tablets
This is a followup of issues like https://github.com/scylladb/scylladb/issues/19504.

In case a keyspace replication-factor prevent a node decommission, there should be a clear error message explaining that.
Currently the error message doesn't hint anything about this to the user:
```
Failed to drain tablets: std::runtime_error (Unable to find new replica for tablet 4eb84450-3084-11ef-993e-55b6485231a6:17 on d105c73b-8556-4ff1-8f4c-102b9eec536a:6 when draining {d105c73b-8556-4ff1-8f4c-102b9eec536a})
```
It would better be something like:
``` 
A decommission is not supported in case a keyspace replication-factor equals the number of nodes in that DC.
Please consider reducing keyspace replication factor in order to have enough replicas
```

### PR DESCRIPTION
**Pull request by Jager227**
to hint user about the reason of that

Fixes #20417

### CODE REVIEW

File: service/tablet_allocator.cc [hunk 1]
```diff
@@ -1601,8 +1601,10 @@ class load_balancer {
                 if (skip) {
                     if (src_node_info.drained && skip->viable_targets.empty()) {
                         auto replicas = tmap.get_tablet_info(source_tablet.tablet).replicas;
-                        throw std::runtime_error(format("Unable to find new replica for tablet {} on {} when draining {} (nodes {}, replicas {})",
-                                                        source_tablet, src, nodes_to_drain, nodes_by_load_dst, replicas));
+                        throw std::runtime_error(
+                            "A decomission is not supported in case a keyspace replication-factor equals the number of nodes in that DC. "
+                            "Please consider reducing keyspace replication factor in order to have enough replicas"
+                        );
```
Comment (tgrabiec) [inline service/tablet_allocator.cc:1607]: We should preserve debug information which was there before. 

You can print "A decomission is not supported ..." only when you examined the state of topology and determined that this is really the cause of this condition. There could be other causes. E.g. removenode, not decommission. Rack overload constraints.

**Review by tgrabiec — CHANGES_REQUESTED** [review]

**Comment by Jager227** [comment]
Damn it, i got some troubles with rebasing. Sorry.

File: service/tablet_allocator.cc (hunk 1 above)
Comment (mykaul) [reply service/tablet_allocator.cc]: @Jager227 - was this review comment addressed in newer version?

File: service/tablet_allocator.cc [hunk 2]
```diff
@@ -1601,8 +1601,16 @@ class load_balancer {
                 if (skip) {
                     if (src_node_info.drained && skip->viable_targets.empty()) {
                         auto replicas = tmap.get_tablet_info(source_tablet.tablet).replicas;
-                        throw std::runtime_error(fmt::format("Unable to find new replica for tablet {} on {} when draining {} (nodes {}, replicas {})",
-                                                        source_tablet, src, nodes_to_drain, nodes_by_load_dst, replicas));
+                        int number_of_replicas = std::size(replicas);
+                        int number_of_nodes = std::size(nodes);
+                        if (number_of_replicas == number_of_nodes) {
+                            throw std::runtime_error(format("Unable to find new replica for tablet {} on {} when draining {}"
```
Comment (tgrabiec) [inline service/tablet_allocator.cc:1607]: "nodes" contains the set of nodes to balance, which may be fewer nodes than the number of nodes in a DC. See 5e28bf9b.

Also "replicas" includes replicas in all DCs, and "nodes" is a subset of nodes in the currently balanced DC. So this logic works only when there is 1 DC in the cluster.

**Review by tgrabiec — CHANGES_REQUESTED** [review]

**Comment by mykaul** [comment]
@Jager227 - are you still working on this?

**Comment by Jager227** [comment]
> @Jager227 - are you still working on this?

No, i moved on other issues

**Pull request closed without merging**